### PR TITLE
feat(guidetooltip): add GuideTooltip component

### PR DESCRIPTION
**Pull request by DanDeMicco**
Screen Shot 2020-06-29 at 1 39 51 PM
Screen Shot 2020-06-29 at 1 24 26 PM
Screen Shot 2020-06-29 at 1 24 23 PM
Screen Shot 2020-06-29 at 1 24 20 PM

### CODE REVIEW

File: src/components/guide-tooltip/GuideTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import './GuideTooltip.scss';
+
+type TooltipProps = Omit<JSX.LibraryManagedAttributes<typeof Tooltip, Tooltip['props']>, 'text' | 'theme'>;
+
+type Props = TooltipProps & {
+    body: React.ReactNode;
+    title?: React.ReactNode;
+    icon?: React.ReactNode;
+    step?: [number, number];
```
Comment (azuruce) [inline src/components/guide-tooltip/GuideTooltip.tsx:17]: should name it steps?

File: src/components/guide-tooltip/GuideTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    secondaryButton?: JSX.LibraryManagedAttributes<typeof Button, Button['props']>;
+};
+
+function GuideTooltip({
+    body,
+    children,
+    className = '',
+    icon,
+    isShown = true,
+    primaryButton,
```
Comment (azuruce) [inline src/components/guide-tooltip/GuideTooltip.tsx:28]: should we name it primaryButtonProps?

File: src/components/guide-tooltip/GuideTooltip.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
+@import '../../styles/variables';
+
+.bdl-GuideTooltip {
+    display: flex;
+}
+
+.bdl-GuideTooltip.bdl-Tooltip {
+    max-width: 408px;
+    padding: 24px 48px 24px 24px;
```
Comment (booc0mtaco) [inline src/components/guide-tooltip/GuideTooltip.scss:9]: can we update these padding/margins to use `bdl-grid-unit`?

File: src/components/guide-tooltip/GuideTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import Button from '../button';
+import messages from './messages';
+
+import './GuideTooltip.scss';
+
+type TooltipProps = Omit<JSX.LibraryManagedAttributes<typeof Tooltip, Tooltip['props']>, 'text' | 'theme'>;
+
+type Props = TooltipProps & {
+    body: React.ReactNode;
+    title?: React.ReactNode;
```
Comment (booc0mtaco) [inline src/components/guide-tooltip/GuideTooltip.tsx:15]: many are optional. What happens if they all are left off? Do we have an example of that?

File: src/components/guide-tooltip/__tests__/GuideTooltip.test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    const title = <div>title</div>;
+    const body = <div>body</div>;
+    const icon = <FolderShared32 />;
+    const step = [1, 3];
+    const primaryButton = { children: 'Next' };
+    const secondaryButton = { children: 'Previous' };
+
+    const getWrapper = props => shallow(<GuideTooltip {...props} />);
+
+    test('should render with title and body', () => {
```
Comment (booc0mtaco) [inline src/components/guide-tooltip/__tests__/GuideTooltip.test.js:17]: an example with none of the optional props may be useful here

File: src/components/guide-tooltip/GuideTooltip.tsx (hunk 1 above)
Comment (DanDeMicco) [reply src/components/guide-tooltip/GuideTooltip.tsx]: updated

File: src/components/guide-tooltip/GuideTooltip.scss (hunk 3 above)
Comment (DanDeMicco) [reply src/components/guide-tooltip/GuideTooltip.scss]: updated

File: src/components/guide-tooltip/GuideTooltip.tsx (hunk 4 above)
Comment (DanDeMicco) [reply src/components/guide-tooltip/GuideTooltip.tsx:15]: This example is the first screenshot. Only had body

File: src/components/guide-tooltip/__tests__/GuideTooltip.test.js (hunk 5 above)
Comment (DanDeMicco) [reply src/components/guide-tooltip/__tests__/GuideTooltip.test.js:17]: Sure, added

**Pull request merged**